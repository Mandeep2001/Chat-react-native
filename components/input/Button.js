import React from 'react';
import { TouchableOpacity, Text, ViewPropTypes, StyleSheet } from 'react-native';
import PropTypes from 'prop-types';
import { Colors, Fonts } from '../../style/styles';

const styles = StyleSheet.create({
  button: { borderRadius: 8, backgroundColor: Colors.primary, alignItems: 'center' },
  text: { ...Fonts.body, color: '#fff', paddingVertical: 10 },
});

const PrimaryButton = ({ text, pressHandler, style }) => {
  return (
    <TouchableOpacity onPress={pressHandler} style={{ ...styles.button, ...style }}>
      <Text style={styles.text}>{text}</Text>
    </TouchableOpacity>
  );
};

PrimaryButton.propTypes = {
  text: PropTypes.string.isRequired,
  pressHandler: PropTypes.func.isRequired,
  style: ViewPropTypes.style,
};

PrimaryButton.defaultProps = { style: {} };

export default PrimaryButton;
