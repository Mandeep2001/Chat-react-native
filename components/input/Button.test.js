import React from 'react';
import { shallow } from 'enzyme';
import Button from './Button';

describe('Button', () => {
  describe('Rendering', () => {
    it('should match to snapshot', () => {
      const component = shallow(
        <Button pressHandler={() => console.log('Premuto')} text="Button" />
      );
      expect(component).toMatchSnapshot();
    });
  });
});
