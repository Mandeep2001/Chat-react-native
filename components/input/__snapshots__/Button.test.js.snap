// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Rendering should match to snapshot 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#33E1ED",
      "borderRadius": 8,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#fff",
        "fontSize": 17,
        "fontWeight": "normal",
        "paddingVertical": 10,
      }
    }
  >
    Button
  </Text>
</TouchableOpacity>
`;
